### ISSUE TITLE
How to assess Reflow on non-web content

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag#1551]
Somewhat related to #1550 , I have had a question from a developer working on a Windows desktop application about how to test and assess Reflow on non-web materials.
The WCAG2ICT material has not been updated since 2013. On the web, one relies on the user agent to provide the zoom functionality. So it is easy to know how to assess.
What is the equivalent on a desktop?
@bruce-usab this applies in Revised 508 to software. Do you have any insight into how people are reporting against it?
@DavidMacDonald you feel that Reflow applies to non-web ICT in your https://www.davidmacd.com/WCAG/wcag2ict-21-discussion.html
Anyone have any idea how to assess?

**Comment by mraccess77** [comment]
I'd recommend using a screen ruler and using the ruler to check each block of content to make sure it fits in the width of 320 device independent pixels.  You may need to resize the application window to assist or use grips,  or other panel/pane features to resize content within the window during the test.

**Comment by awkawk** [comment]
I don't see how we would apply Reflow to non-web software. Non-web documents, sure, but if we are expecting software like Microsoft Word, Acrobat, Photoshop, Zoom, and other desktop applications to meet this SC then I think that we aren't being realistic. 

@mbgower Reflow _does not_ apply in the Revised Section 508 to software. Right now when we test desktop software we respond in the VPAT with Not Applicable.

Should desktop software work correctly down to 320px? That would be great, but this has not been an expectation that has made it to desktop software yet and would require much more more complicated implementation work than is the case for web-based content, so I do not think that it would be appropriate to apply to software yet. Are there any reference implementations for desktop apps that would support it?

**Comment by mraccess77** [comment]
The criterion does not require the window itself resize to 320 - but that for each block of content that the block can fit in that width.  Many aspects of software applications do meet this when you measure at the individual block level.  It is an EN 301 549 requirement.

**Comment by JAWS-test** [comment]
@mraccess77:

> The criterion does not require the window itself resize to 320 - but that for each block of content that the block can fit in that width. 

Regarding web content: This is not how I understand the understanding of 1.4.10. If this is really the case, I think it should definitely be explained in the Understanding. I understand 1.4.10 to mean that in 320px, for example, two columns of text (neither of which is wider than 320px) no longer have to be arranged next to each other, but below each other.

Regarding to non-web software: The WCAG does not apply to this. However, if you have to be compliant with EN 301 549, SC 1.4.10 must also be met for software and documents. I suspect that hardly any document or software meets the criterion. I don't think it makes sense to adopt most WCAG-SC in EN 301 549 for software and documents. I would prefer a criterion for documents and software that text blocks should not be wider than 320px. This is much easier to comply with.

**Comment by detlevhfischer** [comment]
@mraccess77 Hi Jon, I looked up 11.1.4.10 in v.3.1.1 of the EN and do not see where it explicitly calls out 'each block of content'. Is it worth raising the issue with authors of the EN so that they qualify how they think 11.1.4.10 should apply to software?

**Comment by mraccess77** [comment]
EN 301 549 does not call this out - but I believe it would be needed in order for most software to conform.  Yes, it would appear a dialogue is needed to understand how to apply this requirement of EN 301 549 to software.

**Comment by bruce-usab** [comment]
I agree that the EN folks should have qualified how 11.1.4.10 would be applied to software _before_ they adopted 2.2.

**Comment by mbgower** [comment]
> @mbgower Reflow does not apply in the Revised Section 508 to software. Right now when we test desktop software we respond in the VPAT with Not Applicable.

mea culpa, I meant to say it applies in EN. 508R does not include WCAG 2.1, so the new Reflow requirement obviously wouldn't apply.
